MailCore/Cache/MailboxManager/MailboxManager+Thread.swift

```diff
@@ -591,6 +604,21 @@ public extension MailboxManager {
         writableRealm.delete(orphanMessages)
     }
 
+    private func resetFolder(_ folder: Folder) async throws {
+        try writeTransaction { realm in
+            guard let liveFolder = folder.fresh(using: realm) else { return }
+
+            liveFolder.remainingOldMessagesToFetch = Constants.messageQuantityLimit
+            liveFolder.oldMessagesUidsToFetch.removeAll()
+            liveFolder.newMessagesUidsToFetch.removeAll()
+            realm.delete(liveFolder.threads)
+            realm.delete(liveFolder.messages)
```

Comment:
From my tests this correctly removes the threads and messages in the db (not only those in the list)